core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -620,17 +623,12 @@ class KafkaServer(val config: KafkaConfig, time: Time = SystemTime, threadNamePr
     */
   private def getBrokerId: Int =  {
     var brokerId = config.brokerId
-    var logDirsWithoutMetaProps: List[String] = List()
     val brokerIdSet = mutable.HashSet[Int]()
 
     for (logDir <- config.logDirs) {
       val brokerMetadataOpt = brokerMetadataCheckpoints(logDir).read()
-      brokerMetadataOpt match {
-        case Some(brokerMetadata: BrokerMetadata) =>
-          brokerIdSet.add(brokerMetadata.brokerId)
-        case None =>
-          logDirsWithoutMetaProps ++= List(logDir)
-      }
+      if(brokerMetadataOpt.isDefined)
+        brokerIdSet.add(brokerMetadataOpt.get.brokerId)
```

Comment:
Generally `Option.foreach` is better than `isDefined`/`get` (because `get` is generally unsafe and better avoided).